great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1144,7 +1147,10 @@ def execute_split_query(self, split_query: Selectable) -> List[Row]:
             pattern = re.compile(r"(CAST\(EXTRACT\(.*?\))( AS STRING\))", re.IGNORECASE)
             split_query = re.sub(pattern, r"\1 AS VARCHAR)", split_query)
 
-        return self.engine.execute(split_query).fetchall()
+        if isinstance(self.engine, Engine):
+            self.engine = self.engine.connect()
+        query_result: List[Row] = self.engine.execute(split_query).fetchall()
+        return query_result
```

Comment:
What if instead here we use the connection instead of reassigning to self.engine? That would remove the side effect.

```suggestion
        if isinstance(self.engine, Engine):
            connection = self.engine.connect()
        query_result: List[Row] = connection.execute(split_query).fetchall()
        return query_result
```

Reply:
I didn't see this change implemented - would you be able to make this change (in the other part of the file as well) or do we need to set the engine property to a connection? If so can you add a comment that we are overriding with a connection?